desktop_app/src/ui/lib/utils/slash-commands.ts

```diff
@@ -0,0 +1,79 @@
+export enum SlashCommand {
+  CLEAR = '/clear',
+  COMPACT = '/compact',
+}
+
+export interface SlashCommandConfig {
+  command: SlashCommand;
+  description: string;
+  handler?: () => Promise<void> | void;
+}
+
+export interface SlashCommandsConfig {
+  clear: SlashCommandConfig;
+  compact: SlashCommandConfig;
+}
+
+export interface SlashCommandParseResult {
+  isSlashCommand: boolean;
+  command?: SlashCommand;
+  fullCommand?: string;
+  isValidCommand: boolean;
+}
+
+export const AVAILABLE_SLASH_COMMANDS = {
+  clear: {
+    command: SlashCommand.CLEAR,
+    description: 'Clear the current chat and start a new conversation',
+  },
+  compact: {
+    command: SlashCommand.COMPACT,
+    description: 'Summarize the conversation and continue with compacted context',
+  },
+} as const;
```

Comment:
could this just be an array? why do we also need the clear/compact keys here? lastly, why type this `as const`?